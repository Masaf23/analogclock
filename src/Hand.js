import React, { Component } from 'react';
import { ClockHand, ClockHandContainer } from './ClockComponents';
import Util from './Util';

class Hand extends Component {

    constructor(props) {
        super(props);
        this.state = {};
    }

    render() {
        return (
            <ClockHandContainer handAngle={Util.getHandAngle(this.props)} >
                <ClockHand type={this.props.type}
                    handColors={this.props.handColors} />
            </ClockHandContainer>
        )
    }

}

export default Hand;
