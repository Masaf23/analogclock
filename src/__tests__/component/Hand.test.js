import React from "react";
import "jest-styled-components";
import { shallow } from "enzyme";
import "../../setupTests";

import Hand from "../../Hand";

describe("Hand component", () => {
  it("matches second Hand snapshot", () => {
    const wrapper = shallow(
      <Hand type="second" seconds="10" handColors={{ second: "#d81c7a" }} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("matches minute Hand snapshot", () => {
    const wrapper = shallow(
      <Hand type="minute" minutes="10" handColors={{ minute: "#fff" }} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("matches hour Hand snapshot", () => {
    const wrapper = shallow(<Hand type="hour" hours="10" handColors={{ hour: "#fff" }} />);
    expect(wrapper).toMatchSnapshot();
  });
});
