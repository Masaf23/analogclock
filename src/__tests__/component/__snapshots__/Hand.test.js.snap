// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hand component matches hour Hand snapshot 1`] = `
<styled.div
  handAngle={570}
>
  <styled.div
    handColors={
      Object {
        "hour": "#fff",
      }
    }
    type="hour"
  />
</styled.div>
`;

exports[`Hand component matches minute Hand snapshot 1`] = `
<styled.div
  handAngle={330}
>
  <styled.div
    handColors={
      Object {
        "minute": "#fff",
      }
    }
    type="minute"
  />
</styled.div>
`;

exports[`Hand component matches second Hand snapshot 1`] = `
<styled.div
  handAngle={330}
>
  <styled.div
    handColors={
      Object {
        "second": "#d81c7a",
      }
    }
    type="second"
  />
</styled.div>
`;
